app/src/main/java/com/example/myfirstapp/ShoppingTabFragment.java

```diff
@@ -2,16 +2,121 @@
 
 import android.os.Bundle;
 import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.text.Editable;
+import android.text.InputType;
+import android.text.TextWatcher;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
 
 /**
  */
 public class ShoppingTabFragment extends Fragment {
+
+    // Adapter
+    SimpleAdapter recyclerAdapter;
+
+    // RecyclerView
+    RecyclerView recyclerView;
```

Comment:
It's best to keep the visibility as low as possible. So, you might consider making these private as opposed to package-private.